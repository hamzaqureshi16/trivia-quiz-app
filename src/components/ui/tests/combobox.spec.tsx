import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ComboBox } from '../combobox';

describe('Combobox Component', () => {
  const options = [
    {
      value: 'Option 1',
      label: 'Option 1',
    },
    {
      value: 'Option 2',
      label: 'Option 2',
    },
    {
      value: 'Option 3',
      label: 'Option 3',
    },
    {
      value: 'Option 4',
      label: 'Option 4',
    },
    {
      value: 'Option 5',
      label: 'Option 5',
    },
  ];

  it('renders the Combobox with a placeholder', () => {
    render(
      <ComboBox options={options} label="Select an option" value="" onValueChange={value => {
        console.log(value);
      }} />,
    );
    expect(screen.getByText('Select an option')).toBeInTheDocument();
  });
});
