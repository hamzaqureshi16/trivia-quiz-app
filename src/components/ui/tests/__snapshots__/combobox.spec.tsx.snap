// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox Component matches the snapshot 1`] = `
<button
  aria-controls="radix-«r4»"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-[200px] justify-between"
  data-slot="popover-trigger"
  data-state="closed"
  role="combobox"
  type="button"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-chevrons-up-down ml-2 h-4 w-4 shrink-0 opacity-50"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="m7 15 5 5 5-5"
    />
    <path
      d="m7 9 5-5 5 5"
    />
  </svg>
</button>
`;
